src/commands/Moderation/ban.ts

```diff
@@ -25,49 +26,46 @@ export default class extends Command {
 		});
 	}
 
-	async run(message: Message, args: string[]): Promise<void | Message> {
+	async run(context: CommandContext, args: string[]): Promise<void | Message> {
 		//Variable declarations
 		const targetID: Snowflake = args[0].replace(NonDigits, "");
-		let target: any = message.guild?.members.cache.get(targetID);
+		let target: any = context.guild?.members.cache.get(targetID);
 		let reason: string = args.slice(1).join(" ");
 		let notInGuild: boolean = !target;
 		let infID: number = 0;
 
 		if (!notInGuild) {
-			if (await this.client.bulbutils.resolveUserHandle(message, this.client.bulbutils.checkUser(message, target), target.user)) return;
+			if (await this.client.bulbutils.resolveUserHandle(context, this.client.bulbutils.checkUser(context, target), target.user)) return;
```

Comment:
We should look into refactoring in-command permission checks like this out to an override of `Command#validate`. This would enable the command-specific validation steps to be performed during the command handling validation step, and make it easy to use ephemeral replies for invalid uses.

Reply:
Will be moved into `Command#validate` in an upcoming pull request